### ISSUE TITLE
KAKU Asun-650

### ISSUE THREAD

**Issue by Kasper05** [robertklep/homebridge-klikaanklikuit#24]
Is het mogelijk om de Asun-650 aan te sturen met open/dicht.

Bij de "type" kan ik de volgende mogelijkheden invullen: Lightbulb, Outlet, Switch, Fan. Of zijn er nog ander varianten?

**Comment by robertklep** [comment]
Ik denk dat "WindowCovering" het best in de buurt komt, maar in principe maakt het type niet zoveel uit (wordt namelijk alleen gebruikt om tegen HomeKit te zeggen wat voor soort apparaat het is, zodat het er op iOS "goed" uit ziet).

Ik ken het apparaat verder niet, maar als het gewoon op "aan" en "uit" codes reageert zou het moeten werken. Net als bij elk ander apparaat moet je dan wel de juiste codes instellen.

Zo te zien is het een ontvanger die codes kan leren. In dat geval is het 't makkelijkst om het apparaat een nieuwe code te laten leren die specifiek voor Homebridge is.

Hoe dat moet heb ik al eens hier beschreven: https://github.com/robertklep/homebridge-klikaanklikuit/issues/8#issuecomment-313631268 (en de comments die daarop volgen).

Waar het op neerkomt is dat je [`kaku-rpi`](https://github.com/robertklep/node-kaku-rpi) installeert, een standalone tool om codes te versturen. Vervolgens zet je de Asun-650 in "leermodus" zet, en dan stuur je met `kaku-rpi` een unieke code (die je zelf mag verzinnen) naar het apparaat. Die code gebruik je vervolgens ook in `config.json`.

**Comment by Kasper05** [comment]
Bedankt voor de uitleg.
De Asun-650 is nu toegevoegd.
Alleen werkt het nu met de commando's aan/uit. Hoe kan ik dat aanpassen naar open/dicht

**Comment by robertklep** [comment]
> Alleen werkt het nu met de commando's aan/uit. Hoe kan ik dat aanpassen naar open/dicht

Ik kan zo snel niet vinden hoe dat zou moeten. Die "WindowCovering" is ook niet echt handig bij nader inzien, want dat is meer bedoeld voor jaloeziën die meer of minder licht kunnen doorlaten.

De Asun lijkt qua apparaat toch meer op een Switch die aan (open) of uit (dicht) staat.

Ik ben bang dat je het hiermee moet doen.

**Comment by Kasper05** [comment]
Ok. Helaas

Ik vond nog wel dit op het internet

HomeWizard
[https://github.com/rthewhite/homebridge-homewizard/issues/34](url)

**Comment by robertklep** [comment]
> https://github.com/rthewhite/homebridge-homewizard/issues/34

Daar zitten ze een beetje met een vergelijkbaar probleem: HomeWizard beschouwt de Asun-650 als "WindowCovering".

Binnen HomeKit heeft zo'n apparaat heeft geen "open" of "dicht" commando's, maar alleen een percentage (hoeveel licht ie moet doorlaten). En in combinatie met de Asun-650 lijkt me dat niet praktisch.

**Comment by Kasper05** [comment]
Dan is “Switch” het beste. Bedankt voor het meedenken.

**Comment by Kasper05** [comment]
@robertklep 

Heb een paar jaar geleden de vraag gesteld of het mogelijk was om de Asun-650 te bedienen met een "open" of "dicht" commando's. Nu ben ik zelf ook in het bezit van een ICS1000 en heb ik via [https://github.com/djangosoe/homebridge-ics1000-django](url) de plug-in geïnstalleerd. Via deze plugin is het namelijk wel mogelijk de commando's " open" of "dicht" te gebruiken.
Is het idee/mogelijk om deze manier toe te voegen?

**Comment by robertklep** [comment]
@Kasper05 die werkt zo te zien toch ook met `WindowCovering`. Maar het is zo te zien niet triviaal om dat werkend te krijgen (daar komt nog bij dat ik én geen Kaku én geen Homebridge meer gebruik, wat testen lastig maakt).

Als het via de ICS1000-plugin wel werkt kun je die toch blijven gebruiken?

**Comment by Kasper05** [comment]
Klopt, ik kan het via ICS1000 plug-in gebruiken. Ik dacht voor andere mensen die de plug-in van jou willen gebruiken. De ICS1000 wordt namelijk niet meer gemaakt.
